src/Waives.Http/WaivesClient.cs

```diff
@@ -125,6 +125,16 @@ public WaivesClient Login(string clientId, string clientSecret)
         /// <seealso cref="Document.Delete"/>
         public async Task<Document> CreateDocument(Stream documentSource)
         {
+            if (documentSource == null)
+            {
+                throw new ArgumentNullException(nameof(documentSource));
+            }
+
+            if (documentSource.Length < 1)
```

Comment:
:hand: Don't forget to check `.CanSeek` before using `.Length`

Reply:
It might also be an idea to create a `RewindableStream` which allows you to read a byte (to check that it's not empty), buffer it, and then rewind it (but without having to read the whole thing into memory as a MemoryStream).

Reply:
I've updated this PR to call `CanSeek` as suggested. If the stream is not seekable, then we wrap it in a `FileBufferingReadStream` (a class pulled in from ASP.NET Core), read a single byte and rewind that stream to the beginning. The wrapping `FileBufferingReadStream` is what is then used as the document stream for the Create Document request.